alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -69,47 +71,82 @@ class DocLevelMonitorQueries(private val client: AdminClient, private val cluste
-                        override fun onResponse(response: AcknowledgedResponse) {
-                            log.info("Percolation index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX} updated with new mappings")
-
-                            val request = BulkRequest().setRefreshPolicy(refreshPolicy).timeout(indexTimeout)
-
-                            queries.forEach {
-                                var query = it.query
-
-                                properties.forEach { prop ->
-                                    query = query.replace("${prop.key}:", "${prop.key}_$monitorId:")
-                                }
-                                val indexRequest = IndexRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
-                                    .id(it.id + "_$monitorId")
-                                    .source(mapOf("query" to mapOf("query_string" to mapOf("query" to query)), "monitor_id" to monitorId))
-                                request.add(indexRequest)
-                            }
-
-                            queryClient.bulk(request, docLevelQueryIndexListener)
-                        }
 
-                        override fun onFailure(e: Exception) {
-                            if (indexMonitorActionListener != null) {
-                                indexMonitorActionListener.onFailure(AlertingException.wrap(e))
-                            } else executeMonitorActionListener?.onFailure(AlertingException.wrap(e))
+        val getAliasesRequest = GetAliasesRequest(index)
+        queryClient.admin().indices().getAliases(
+            getAliasesRequest,
+            object : ActionListener<GetAliasesResponse> {
+                override fun onResponse(getAliasesResponse: GetAliasesResponse?) {
+                    val aliasIndices = getAliasesResponse?.aliases?.keys()?.map { it.value }
+                    val isAlias = aliasIndices != null && aliasIndices.isNotEmpty()
+                    val indices = if (isAlias) getAliasesResponse?.aliases?.keys()?.map { it.value } else listOf(index)
+                    val indexRequests = mutableListOf<IndexRequest>()
+                    log.info("indices: $indices")
+                    indices?.forEach { indexName ->
+                        if (clusterState.routingTable.hasIndex(indexName)) {
+                            val indexMetadata = clusterState.metadata.index(indexName)
+
+                            if (indexMetadata.mapping() != null) {
+                                log.info("Index name: $indexName")
+                                val properties = (
+                                    (indexMetadata.mapping()?.sourceAsMap?.get("properties"))
+                                        as Map<String, Map<String, Any>>
+                                    )
+                                val updatedProperties = properties.entries.associate { "${it.key}_$monitorId" to it.value }.toMutableMap()
```

Comment:
this will result in a conflict if two or more indices have the same property name. 
the conflict can be resolved by adding index keys to the property names.
instead of `${it.key}_$monitorId`, we can use this `${it.key}_${indexName}_$monitorId`

Reply:
done